backend/TraderX/src/main/java/cmpe451/group6/rest/article/controller/ArticleController.java

```diff
@@ -0,0 +1,117 @@
+package cmpe451.group6.rest.article.controller;
+
+import cmpe451.group6.Util;
+import cmpe451.group6.rest.article.dto.ArticleDTO;
+import cmpe451.group6.rest.article.dto.ArticleInfoDTO;
+import cmpe451.group6.rest.article.model.Article;
+import cmpe451.group6.rest.article.service.ArticleService;
+import io.swagger.annotations.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.Date;
+import java.util.List;
+
+import static cmpe451.group6.authorization.exception.GlobalExceptionHandlerController.GENERIC_ERROR_RESPONSE;
+
+@RestController
+@RequestMapping("/article")
+@Api(tags = "Article")
+public class ArticleController {
+
+    @Autowired
+    private ArticleService articleService;
+
+    @Autowired
+    private Util util;
+
+    @GetMapping(value = "/all")
+    @ResponseStatus(HttpStatus.OK)
+    @PreAuthorize("hasRole('ROLE_ADMIN') or hasRole('ROLE_BASIC') or hasRole('ROLE_TRADER')")
+    @ApiOperation(value = "Gets all articles (public profiles and private but following). ", response = ArticleDTO.class)
+    @ApiResponses(value = { @ApiResponse(code = 400, message = GENERIC_ERROR_RESPONSE) })
+    public List<ArticleDTO> getArticles(HttpServletRequest req) {
+        return articleService.getArticles(util.unwrapUsername(req));
+    }
+
+    @GetMapping(value = "/byUsername/{username}")
+    @ResponseStatus(HttpStatus.OK)
+    @PreAuthorize("hasRole('ROLE_ADMIN') or hasRole('ROLE_BASIC') or hasRole('ROLE_TRADER')")
+    @ApiOperation(value = "Gets all articles of user (public or private but following). ", response = ArticleDTO.class)
+    @ApiResponses(value = { @ApiResponse(code = 400, message = GENERIC_ERROR_RESPONSE) })
```

Comment:
Can we add `PreconditionFailed` response here and other necessary endpoints?

Reply:
I have added PrecoditionFailed api responses.